app/input_objects/pages/secondary_skip_input.rb

```diff
@@ -55,25 +55,43 @@ def routing_page_options
     pages_after_current_page(FormRepository.pages(form), page).map { |p| OpenStruct.new(id: p.id, question_text: p.question_with_number) }
   end
 
+  def end_page_name
+    I18n.t("page_route_card.check_your_answers")
+  end
+
+  def answer_value
+    page.routing_conditions.find { |rc| rc.answer_value.present? }.answer_value
+  end
+
+  def primary_route_goto_page_name
+    primary_route = page.routing_conditions.find { |rc| rc.answer_value.present? }
+
+    if primary_route.skip_to_end?
+      return I18n.t("page_route_card.check_your_answers")
+    end
+
+    question_name(primary_route.goto_page_id) || I18n.t("page_route_card.goto_page_invalid")
+  end
+
   def question_name(page_id)
-    target_page = FormRepository.pages(form).find { |page| page.id == page_id }
+    target_page = form.pages.find { |page| page.id == page_id }
```

Comment:
We want to make sure we're using the repository whenever we might be doing a network request.

Reply:
another good spot.

I've changed it to use the repository. I'm going to look into memoizing the form's pages in the secondary_skip_input to make sure we are not making   loads of network calls we don't need.